adafruit_tinylora/adafruit_tinylora.py

```diff
@@ -281,11 +281,13 @@ def send_packet(self, lora_packet, packet_length):
         self._write_u8(_REG_OPERATING_MODE, _MODE_TX)
         # wait for TxDone IRQ
         print('Sending packet')
-        while not self._irq.value:
+        send_attempt = 0
+        while not self._irq.value and send_attempt < 15:
             # waiting for TxDone
-            print('.', end='')
```

Comment:
This waits at lease one second - I think in most cases the send time is < 1 second so it wastes time. It also fails silently.  I suggest using time.monotonic() to set a known timeout  - possibly setable as a keyword argument and then throw and error if the timeout is reached. There is an example in the rfm9x library - https://github.com/adafruit/Adafruit_CircuitPython_RFM9x/blob/master/adafruit_rfm9x.py#L536

Reply:
configurable timeout is a good idea, added to `send_data` method